spec/unit/recipes/site_pr_builder_spec.rb

```diff
@@ -0,0 +1,56 @@
+require_relative '../../spec_helper'
+
+describe 'osl-jenkins::site_pr_builder' do
+  ALL_PLATFORMS.each do |p|
+    context "#{p[:platform]} #{p[:version]}" do
+      cached(:chef_run) do
+        ChefSpec::SoloRunner.new(p) do |node|
+          node.set['osl-jenkins']['credentials']['git'] = {
+            'site_pr_builder' => {
+              user: 'manatee',
+              token: 'token_password'
+            }
+          }
+          node.set['osl-jenkins']['credentials']['jenkins'] = {
+            'site_pr_builder' => {
+              user: 'manatee',
+              pass: 'password',
+              trigger_token: 'trigger_token'
+            }
+          }
+        end.converge(described_recipe)
+      end
+
+      include_context 'common_stubs'
+      it 'converges successfully' do
+        expect { chef_run }.to_not raise_error
+      end
+
+      sites = %w(beaver-barcamp-pelican_pr_builder cass-pelican_pr_builder
+                 osuosl-pelican_pr_builder wiki_pr_builder docs_pr_builder)
+
+      sites.each do |job|
+        it do
+          expect(chef_run).to create_directory("/var/chef/cache/#{job}").with(recursive: true)
+        end
+      end
+
+      sites.each do |job|
+        it do
+          expect(chef_run).to create_jenkins_job(job)
+            .with(config: "/var/chef/cache/#{job}/config.xml")
+        end
+      end
+
+      sites.each do |job|
+        it do
+          expect(chef_run).to create_template("/var/chef/cache/#{job}/config.xml")
+            .with(
+              source: 'site_pr_builder_config.xml.erb',
+              mode: 0440
+            )
+        end
+      end
```

Comment:
You might as well do all three of these loops as one loop which includes all three of these tests.